tests/test_precision.py

```diff
@@ -38,3 +38,20 @@ def test_mpdist_snippets_s():
     npt.assert_almost_equal(
         ref_fractions, cmp_fractions, decimal=config.STUMPY_TEST_PRECISION
     )
+
+
+def test_distace_profile():
+    # This test function raises an error when the distance profile between
+    # the query `Q = T[i: i+m]`  and `T` becomes non-zero at index `i`.
```

Comment:
According to this comment, we may change the test function to something like this:

```
npt.assert_almost_equal(0.0, D_comp[i])
```

@seanlaw 
What do you think?

Reply:
Isn't `npt.assert_almost_equal(D_ref, D_comp)` already a superset of `npt.assert_almost_equal(0.0, D_comp[i])`? If not, maybe you can assert both?

Reply:
Yeah...but I had two reasons in mind:
(1) we are just focusing on testing wether the distance to self is 0 or not. However, now that I think more about it, testing all the values of the array is better as it ensures us that we are not messing with the output when `query_idx` is provided.

(2) Fortunately, we can trust naive function for our case as naive version gives 0 for self-match distances. But using 0.0 is more direct and does not need us to know whether naive gives `0.0` for the self-match distances or not.